### DISCUSSION TITLE
Proper way to programmatically open the Traits/Style Managers' panels?

### DISCUSSION THREAD

**Discussion by SuperDelphi** [GrapesJS/grapesjs#5910] [Q&A] [answered]
Hello!

In my GrapesJS project, I've defined two keymaps to open the Traits Manager and the Style Manager, respectively.
I first tried to bind the keymaps to ``open-tm`` and ``open-sm``, but it doesn't work as intended (doesn't close the previously opened panel and doesn't set the associated button as active, visually).

I'm currently doing this:
```ts
editor.Panels.getButton('views', 'open-tm').set('active', true);
```

It seems like I emulate a button click (which looks like a hack), so my question is the following: Is it the proper way to open a panel?

Thanks in advance!

**Comment by artf** [comment]
If you try your code with the default grapesjs it works as expected so I guess it might be related to your customizations.

**Reply by SuperDelphi** [reply]
> If you try your code with the default grapesjs it works as expected so I guess it might be related to your customizations.

Thanks for your answer. However what I said is that the code works well, but just using ``editor.runCommand('open-tm')`` doesn't close the other panels nor update the button bound to the panel (which can be understandable). Is this normal behaviour?

**Reply by artf** [reply, marked as answer]
> Is this normal behaviour?

Of course, those are 2 different things, Command API and Panel API.
If you want the panel/button to react you have to use Panel API, which triggers the Command API but not the opposite way, otherwise it would be a total mess.

**Reply by SuperDelphi** [reply]
Thank you!
